File: foxy/parent/src/command/vanilla/entertainment/FateExecutor.ts
import { bot } from "../../../FoxyLauncher";
import UnleashedCommandExecutor from "../../structures/UnleashedCommandExecutor";
import { ExtendedUser } from "../../../structures/types/DiscordUser";

export default class FateExecutor {
    async execute(context: UnleashedCommandExecutor, endCommand, t) {
        const user = await context.getOption<ExtendedUser>('user', 'users');

        if (!user) {
            context.reply({
                content: context.makeReply(bot.emotes.FOXY_SCARED, t('commands:global.noUser'))
            });

            return endCommand();
        }

        const list = [
            t('commands:fate.couple'),
            t('commands:fate.friend'),
            t('commands:fate.lover'),
            t('commands:fate.enemy'),
            t('commands:fate.sibling'),
            t('commands:fate.parent'),
            t('commands:fate.married')
        ]

        const rand = list[Math.floor(Math.random() * list.length)];
        await context.reply({
            content: context.makeReply(bot.emotes.FOXY_YAY, t('commands:fate.result', { user: (await context.getAuthor()).id.toString(), fate: rand, mention: user.id.toString() }))
        });

        endCommand();
    }
}

File: foxy/src/main/kotlin/net/cakeyfox/foxy/interactions/vanilla/entertainment/FateExecutor.kt
package net.cakeyfox.foxy.interactions.vanilla.entertainment

import net.cakeyfox.common.FoxyEmotes
import net.cakeyfox.foxy.interactions.commands.CommandContext
import net.cakeyfox.foxy.interactions.commands.UnleashedCommandExecutor
import net.cakeyfox.foxy.interactions.pretty
import net.dv8tion.jda.api.entities.User

class FateExecutor : UnleashedCommandExecutor() {
    override suspend fun execute(context: CommandContext) {
        val user = context.getOption("user", 0, User::class.java)
        if (user == null) return

        val fateList = listOf(
            context.locale["fate.couple"],
            context.locale["fate.friend"],
            context.locale["fate.enemy"],
            context.locale["fate.soulmate"],
            context.locale["fate.crush"],
            context.locale["fate.sibling"],
            context.locale["fate.parent"],
            context.locale["fate.married"],
        )

        val fate = fateList.random()

        context.reply {
            content = pretty(
                FoxyEmotes.FoxyYay,
                context.locale["fate.onAParallelUniverse", user.asMention, fate]
            )
        }
    }
}